### DISCUSSION TITLE
Q-Input validate() Function not working as expected

### DISCUSSION THREAD

**Discussion by MrMarlin** [quasarframework/quasar#9022] [General - Components / Directives / etc]
Hello fellow coders

I am currently having an issue with the q-input validation function. I have a q-input which gets its `:rules` from a prop. On `@input` I have a function which triggers the validation of this field and makes a call to another function if the result of the validation is `true`. My Problem now is, that the result from the built-in `validate()` function is always one validation "behind". Let me explain.

```
<template>
  <q-input
    no-error-icon
    v-model="parameter[name]"
    :label="label"
    filled
    :type="type"
    @input="validateTextField"
    ref="textInput"
    :rules="[rule]"
    style="width: 100%"
  />
</template>
<script>
import Vue from "vue";
export default {
  name: "textdefinition",
  props: {
    path: String,
    label: String,
    name: String,
    type: String,
    rule: Function,
    updateCallback: Function
  },
  data() {
    return {
      parameter: {}
    };
  },
  methods: {
    validateTextField() {
      let result = this.$refs.textInput.validate();
      console.log(result);
      if (result) {
        this.updateCallback(this.path, this.name, this.parameter[this.name]);
      }
    }
  },
  mounted() {
    Vue.set(this.parameter, this.name, "");
  }
};
</script>
```

I have a rule on the q-input `return value.length >= 3 || 'Input must at least be three characters long.'` set via a prop. In the UI this works as expected and the input is validated correctly. But if I save the result of the q-input's `validate()` method in the `result` variable and print it after, I get wrong result on the "edge cases". 

Pseudo output of `console.log(result)`:
```
Input            Output
t                False
te               False
tes              False
test             True
tes              True
te               True
t                False
```

Does anybody have an idea what I am doing wrong or if this is a bug? 

Thanks and all the best

**Comment by metalsadman** [comment]
Show code how you are using it, better make a codesandbox
